app/services/ApprovalService.scala

```diff
@@ -31,24 +31,28 @@ import scala.concurrent.Future
 class ApprovalService @Inject() (repository: ApprovalRepository) {
 
   val logger: Logger = Logger(this.getClass)
+  implicit val apFormat: Format[ApprovalProcess] = ApprovalProcessFormatter.mongoFormat
 
-  def save(process: JsObject): Future[RequestOutcome[String]] = {
+  def save(approvalProcess: JsObject): Future[RequestOutcome[String]] = {
 
-    def saveProcess(id: String): Future[RequestOutcome[String]] = repository.update(id, process) map {
-      case Left(_) => Left(Errors(InternalServiceError))
-      case result => result
+    def saveProcess(process: ApprovalProcess): Future[RequestOutcome[String]] = {
+      repository.update(process.meta.id, process) map {
+        case Left(_) => Left(Errors(InternalServiceError))
+        case result => result
+      }
     }
 
-    process.validate[Process] match {
-      case JsSuccess(process, _) => saveProcess(process.meta.id)
+    approvalProcess.validate[ApprovalProcess] match {
+      case JsSuccess(process, _) =>
+        saveProcess(process)
       case JsError(errors) =>
         logger.error(s"Parsing process failed with the following error(s): $errors")
         Future { Left(Errors(BadRequestError)) }
     }
 
   }
 
-  def getById(id: String): Future[RequestOutcome[JsObject]] = {
+  def getById(id: String): Future[RequestOutcome[ApprovalProcess]] = {
```

Comment:
as earlier, match data format of save

Reply:
As per discussion, leaving as is

Reply:
changed method to return the embedded process JsObject